source/extensions/tracers/zipkin/zipkin_core_types.h

```diff
@@ -479,7 +479,10 @@ class Span : public ZipkinBase {
   /**
    * @return the span's trace id as a hexadecimal string.
    */
-  const std::string traceIdAsHexString() const { return Hex::uint64ToHex(trace_id_); }
+  const std::string traceIdAsHexString() const {
+    return (trace_id_high_.has_value() ? Hex::uint64ToHex(trace_id_high_.value()) : "") +
```

Comment:
I don't know cpp, but guessing the conditional would be quicker made in a way that doesn't always cat values? 

ex.

```
return trace_id_high_.has_value()
           ? Hex::uint64ToHex(trace_id_high_.value()) +  Hex::uint64ToHex(trace_id_)
           :  Hex::uint64ToHex(trace_id_);
```